app/src/main/java/org/simple/clinic/analytics/Analytics.kt

```diff
@@ -27,20 +28,26 @@ object Analytics {
     reporters = reporters - reporter
   }
 
-  fun setUserId(uuid: UUID) {
+  fun setLoggedInUser(user: User) {
     reporters.forEach {
-      it.safely("Error setting user ID!") {
-        val uuidString = uuid.toString()
+      it.safely("Error setting logged in user!") {
+        setLoggedInUser(user, false)
+      }
+    }
+  }
 
-        setUserIdentity(uuidString)
+  fun setNewlyRegisteredUser(user: User) {
+    reporters.forEach {
+      it.safely("Error setting newly registered user!") {
```

Comment:
What does `safely` do?

Reply:
Analytics is something that is not crucial to using the app. In the unlikely case that reporting anything to analytics throws an exception for whatever reason, the app must proceed as if nothing has happened.

This block is designed to swallow that error and just report it to Sentry (or the crash reporting service).

I realise now that it can be done in a better way by having a decorator `AnalyticsReporter` that swallows the error and reports it to crash reporting instead. I'll make this change.